src/lib/react/Grid/index.tsx

```diff
@@ -0,0 +1,186 @@
+        ({ ar }: { ar: number }) => {
+            if (ar !== cell.aspectRatio) {
+                onSetAspectRatio({ aspectRatio: ar });
+            }
+        },
+        [cell.aspectRatio, onSetAspectRatio]
+    );
+
+    return (
+        <div
+            style={{
+                position: "absolute",
+                width: cell.bounds.width,
+                height: cell.bounds.height,
+                boxSizing: "border-box",
+                top: cell.origin.top,
+                left: cell.origin.left,
+            }}
+        >
+            {render ? (
+                render()
+            ) : participant.stream ? (
+                <VideoView
+                    stream={participant.stream}
+                    onSetAspectRatio={({ aspectRatio }) => handleAspectRatioChange({ ar: aspectRatio })}
+                    onResize={onResize as any}
+                />
+            ) : null}
+        </div>
+    );
+}
+
+interface GridProps {
+    roomConnection: RoomConnectionRef;
+    renderParticipant?: ({
+        cell,
+        participant,
+    }: {
+        cell: { clientId: string; bounds: Bounds; origin: Origin };
+        participant: RemoteParticipant | LocalParticipant;
+    }) => React.ReactNode;
+    videoGridGap?: number;
+}
+
+function Grid({ roomConnection, renderParticipant, videoGridGap = 0 }: GridProps) {
+    const { remoteParticipants, localParticipant } = roomConnection.state;
+    const gridRef = React.useRef<HTMLDivElement>(null);
+    const [containerFrame, setContainerFrame] = React.useState<Frame | null>(null);
+    const [aspectRatios, setAspectRatios] = React.useState<{ clientId: string; aspectRatio: number }[]>([]);
+
+    // Calculate container frame on resize
+    React.useEffect(() => {
+        if (!gridRef.current) {
+            return;
+        }
+
+        const resizeObserver = new ResizeObserver(
+            debounce(
+                () => {
+                    setContainerFrame(
+                        makeFrame({
+                            width: gridRef.current?.clientWidth,
+                            height: gridRef.current?.clientHeight,
+                        })
+                    );
+                },
+                { delay: 60 }
+            )
+        );
+        resizeObserver.observe(gridRef.current);
+
+        return () => {
+            resizeObserver.disconnect();
+        };
+    }, []);
+
+    // Merge local and remote participants
+    const participants = React.useMemo(() => {
+        return [...(localParticipant ? [localParticipant] : []), ...remoteParticipants];
+    }, [remoteParticipants, localParticipant]);
+
+    // Make video cells
+    const videoCells = React.useMemo(() => {
+        return participants.map((participant) => {
+            const aspectRatio = aspectRatios.find((item) => item.clientId === participant?.id)?.aspectRatio;
+
+            return makeVideoCellView({
+                aspectRatio: aspectRatio ?? 16 / 9,
```

Comment:
Why `??` here?

Reply:
`aspectRatio` is possibly `undefined` here. I'm looking for it in the list of aspectRatios, and it could not be set for this specific participant at this point. So the `??` is just a guard, to set 16/9 as the default